bindings/node/test/clientEncryption.test.js

```diff
@@ -900,44 +902,59 @@ describe('ClientEncryption', function () {
           encryptedFields: { fields: ['not an array', { keyId: keyId }, { keyId: {} }] }
         });
       });
+    });
 
-      it('only passes options.masterKey to createDataKey', async () => {
-        const masterKey = Symbol('key');
-        const createDataKey = sinon
-          .stub(clientEncryption, 'createDataKey')
-          .resolves(new Binary(Buffer.alloc(16, 0)));
-        const result = await clientEncryption.createEncryptedCollection(db, collectionName, {
-          provider: 'aws',
-          createCollectionOptions: { encryptedFields: { fields: [{}] } },
-          masterKey
-        });
-        expect(result).to.have.property('collection');
-        expect(createDataKey).to.have.been.calledOnceWithExactly('aws', { masterKey });
+    it('only passes options.masterKey to createDataKey', async () => {
+      const masterKey = Symbol('key');
+      const createDataKey = sinon
+        .stub(clientEncryption, 'createDataKey')
+        .resolves(new Binary(Buffer.alloc(16, 0)));
+      const result = await clientEncryption.createEncryptedCollection(db, collectionName, {
+        provider: 'aws',
+        createCollectionOptions: { encryptedFields: { fields: [{}] } },
+        masterKey
       });
+      expect(result).to.have.property('collection');
+      expect(createDataKey).to.have.been.calledOnceWithExactly('aws', { masterKey });
     });
 
-    context('when subroutines throw the error is wrapped', () => {
-      it('throws MongoCryptCreateDataKeyError if createDataKey rejects', async () => {
-        const customError = new Error('evil!');
+    context('when createDataKey rejects', () => {
+      const customError = new Error('evil!');
+      const keyId = new Binary(Buffer.alloc(16, 0), 4);
+      const createCollectionOptions = {
+        encryptedFields: { fields: [{}, {}, { keyId: 'cool id!' }] }
+      };
+      const createDataKeyRejection = async () => {
         const stub = sinon.stub(clientEncryption, 'createDataKey');
-        const keyId = new Binary(Buffer.alloc(16, 0), 4);
         stub.onCall(0).resolves(keyId);
         stub.onCall(1).rejects(customError);
         stub.onCall(2).resolves(keyId);
 
-        const createCollectionOptions = {
-          encryptedFields: { fields: [{}, {}, { keyId: 'cool id!' }] }
-        };
         const error = await clientEncryption
```

Comment:
any reason to not inline this into each test case?

Reply:
Like copy this to each it block? I put it here b/c it's used in an assertion in the last it block

Reply:
from an organization standpoint, it might make more sense to

- put all the stubbing into a beforeEach hook
- each test then just calls `await clientEncryption.createEncryptedCollection`

no need to call a helper method in the test instead of calling the method directly